src/features/device-state/routes/state-v3.ts

```diff
@@ -0,0 +1,437 @@
+import type { RequestHandler } from 'express';
+import type { Request } from 'express';
+
+import * as _ from 'lodash';
+import {
+	captureException,
+	handleHttpErrors,
+} from '../../../infra/error-handling';
+import {
+	filterDeviceConfig,
+	formatImageLocation,
+	getReleaseForDevice,
+	serviceInstallFromImage,
+	setMinPollInterval,
+} from '../utils';
+import { sbvrUtils, errors } from '@balena/pinejs';
+import { events } from '..';
+import { Expand } from 'pinejs-client-core';
+
+const { UnauthorizedError } = errors;
+const { api } = sbvrUtils;
+
+export type EnvVarList = Array<{ name: string; value: string }>;
+export const varListInsert = (
+	varList: EnvVarList,
+	obj: Dictionary<string>,
+	filterFn: (name: string) => boolean = () => true,
+) => {
+	varList.forEach(({ name, value }) => {
+		if (filterFn(name)) {
+			obj[name] = value;
+		}
+	});
+};
+export const rejectUiConfig = (name: string) =>
+	!/^(BALENA|RESIN)_UI/.test(name);
+
+type LocalStateApp = StateV3['local']['apps'][string];
+type ServiceComposition = AnyObject;
+export type StateV3 = {
+	[uuid: string]: {
+		name: string;
+		parent_device?: string;
+		apps: {
+			[uuid: string]: {
+				/**
+				 * @deprecated to be removed in state v4
+				 */
+				id: number;
+				name: string;
+				parent_app?: string;
+				is_host?: boolean;
+				config: {
```

Comment:
I thought we were keeping this config on the device level

Reply:
You're right, I'll get this fixed